docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
 
     <h4><a id="streams_dsl_source" href="#streams_dsl_source">Create Source Streams from Kafka</a></h4>
 
-    <p>
-        Either a <b>record stream</b> (defined as <code>KStream</code>) or a <b>changelog stream</b> (defined as <code>KTable</code> or <code>GlobalKTable</code>)
-        can be created as a source stream from one or more Kafka topics (for <code>KTable</code> and <code>GlobalKTable</code> you can only create the source stream
-        from a single topic).
-    </p>
-
-    <pre class="brush: java;">
-    KStreamBuilder builder = new KStreamBuilder();
+    <table class="data-table" border="1">
+  <tbody><tr>
+          <th>Reading from Kafka</th>
+          <th>Description</th>
+  </tr>
+      <tr>
+        <td><b>Stream</b>: input topic(s) -> <code>KStream</code></td>
+        <td>Create a <code>KStream</code> from the specified Kafka input topic(s), interpreting the data as a record stream.
+	  A <code>KStream</code> represents a partitioned record stream. 
+	  <p>
+	  Slightly simplified, in the case of a KStream, the local KStream instance of every application instance will be populated
+	  with data from only a subset of the partitions of the input topic. Collectively, i.e. across all application instances,
+	  all the partitions of the input topic will be read and processed.
+	  <p>
+	  <pre class="brush: java;">
+	  import org.apache.kafka.common.serialization.Serdes;
+	  import org.apache.kafka.streams.kstream.KStreamBuilder;
+	  import org.apache.kafka.streams.kstream.KStream;
+
+	  KStreamBuilder builder = new KStreamBuilder();
+
+	  KStream&lt;String, Long&gt; wordCounts = builder.stream(
+	  Serdes.String(), /* key serde */
+	  Serdes.Long(),   /* value serde */
+	      "word-counts-input-topic" /* input topic */);
+	      </pre>
+	  When to provide serdes explicitly:
+	  <ul>
+	  <li>If you do not specify serdes explicitly, the default serdes from the configuration are used.</li>
```

Comment:
There is no configuration section yet. We'll have to do this in a second pass (this is a problem with doing this piece-wise).